src/sqlfluff/dialects/dialect_athena.py

```diff
@@ -643,3 +644,31 @@ class IntervalExpressionSegment(BaseSegment):
             ),
         ),
     )
+
+
+class ShowStatementSegment(BaseSegment):
+    """A `show` execute statement.
+
+    Full Apache Hive `SHOW` reference:
+    https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Show
+
+    Athena supported subset:
+    https://docs.aws.amazon.com/athena/latest/ug/ddl-reference.html
+    """
+
+    type = "show_statement"
+    match_grammar = Sequence(
+        "SHOW",
+        OneOf(
+            Sequence(
+                "TABLES",
+                Sequence("IN", Ref("TableReferenceSegment"), optional=True),
```

Comment:
I think both of these `TableReferenceSegment` should be `DatabaseReferenceSegment`